crates/core/component/funding/src/component/liquidity_tournament/nullifier/mod.rs

```diff
@@ -0,0 +1,50 @@
+use async_trait::async_trait;
+use penumbra_sdk_txhash::TransactionId;
+
+use crate::component::state_key;
+use cnidarium::{StateRead, StateWrite};
+use penumbra_sdk_proto::{StateReadProto, StateWriteProto};
+use penumbra_sdk_sct::{component::clock::EpochRead, Nullifier};
+
+#[allow(dead_code)]
+#[async_trait]
+pub trait NullifierRead: StateRead {
+    /// Returns the `TransactionId` if the nullifier has been spent; otherwise, returns None.
+    async fn get_lqt_spent_nullifier(&self, nullifier: Nullifier) -> Option<TransactionId> {
+        // Grab the ambient epoch index.
+        let epoch_index = self
+            .get_current_epoch()
+            .await
+            .expect("epoch is always set")
+            .index;
+
+        let nullifier_key = &state_key::lqt::v1::nullifier::key(epoch_index, &nullifier);
+
+        let tx_id: Option<TransactionId> = self
+            .nonverifiable_get(&nullifier_key.as_bytes())
+            .await
+            .expect("infallible");
```

Comment:
```suggestion
            .expect("failed to retrieve key {} from non-verifiable storage", &nullifier_key);
```

Let's add a bit more context here.